### PR TITLE
Added EthersAPI file (WIP) and error checking on proposal/grant applications

### PR DESCRIPTION
**Pull request by CptAstro**
In this PR I added an EthersAPI file that will contain all of our Ethers library functionality. For now I just have two functions in there to be used for testing, but more will be added as I work towards implementing this API with our frontend. @jagger-harris can also use this file to store any functions he will use to interact with Ethers and the blockchain. This file is an ongoing work in process.

I also added a quick error box on the Proposal and Grant application forms that'll display if not all fields are filled out. This is a super basic error checking component to ensure that incomplete proposals don't get uploaded to IPFS. 

Note: It says there are 19 total commits, but everything from "Added IPFS functionality..." up until "Added error box..." is already merged into our upstream repo, so I'm not sure why it thinks there's more commits than there really are. May just be a git bug - I've already double checked to make sure all changes mentioned are already pushed.

### CODE REVIEW

File: frontend/src/data/EthersApi.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+async function main() {
+    await mintAndDelegate();
+    let proposalIds = await generateMockProposals();
+    await simulateVoting(proposalIds);
+    let props = await getBlockchainData(proposalIds);
+    console.log(props);
+}
+
+main();
```
Comment (victor-wei126) [inline frontend/src/data/EthersApi.js:139]: Nitpick: add newline to end of files

File: frontend/src/data/EthersApi.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    // Get all 20 accounts generated by Hardhat
+    let accounts;
+    accounts = await ethers.getSigners();
+
+    // Loop through all the accounts, give each one 1 NFT + delegate so they get voting power
+    for(let a=0; a<accounts.length; a++){
+        await membershipNFT.safeMint(accounts[a].address);
+        await membershipNFT.connect(accounts[a]).delegate(accounts[a].address);
+    }
+}
```
Comment (victor-wei126) [inline frontend/src/data/EthersApi.js:29]: Do you plan for all the functions in this file to be test functions? How do you plan to separate out the test functions vs. the actual API functions?

File: frontend/src/pages/GrantsApp.js [hunk 3]
```diff
@@ -66,6 +69,7 @@ function GrantsApp () {
                         <input type="text" name="GrantAmount" onChange={e => setAmount(e.target.value)} class="form-control" id="exampleFormControlInput1" placeholder="How money money is needed for your project?"></input>
                     </div>
                     <button class="btn btn-primary" type="button" onClick={submitApp}>Submit Grant</button>
+                    {error && <AppError />}
```
Comment (victor-wei126) [inline frontend/src/pages/GrantsApp.js:72]: Does the error reflect something on the UI? I'm guessing the component `<AppError />` contains that graphic?

**Review by victor-wei126 — APPROVED** [review]
LGTM, besides some very minor comments. Feel free to merge after addressing those.

File: frontend/src/pages/GrantsApp.js (hunk 3 above)
Comment (CptAstro) [reply frontend/src/pages/GrantsApp.js:72]: Yup, it's just a super basic error message the pops up on the screen. We could change this to a modal or something else in the future

File: frontend/src/data/EthersApi.js (hunk 2 above)
Comment (CptAstro) [reply frontend/src/data/EthersApi.js:29]: Right now I'm just creating it as one file so I can make sure I have everything that I need. Once I start integrating with React (likely my next PR) I'll separate it into two files, EthersAPI.js and EthersTestAPI.js or something along those lines. In this current PR, getBlockchainData() is the only function that will be used both in testing and in production.

**Pull request merged**